typescript/src/purchaseParams.ts

```diff
@@ -0,0 +1,38 @@
+import {
+  GoogleProductChangeInfo,
+  PurchasesPackage, PurchasesPromotionalOffer,
+  PurchasesStoreProduct,
+  PurchasesWinBackOffer,
+  SubscriptionOption
+} from "./offerings";
+
+/**
+ * Holds parameters to initialize a purchase in our SDK.
+ * @public
+ */
+export interface PurchaseParams {
+  /**
+   * The {@link PurchasesPackage}, {@link PurchasesStoreProduct} or {@link SubscriptionOption} to purchase.
+   */
+  itemToPurchase: PurchasesPackage | PurchasesStoreProduct | SubscriptionOption;
+  /**
+   * Google Play only. Optional {@link GoogleProductChangeInfo} you
+   * wish to upgrade from containing the oldProductIdentifier and the optional prorationMode.
+   */
+  googleProductChangeInfo?: GoogleProductChangeInfo | null;
+  /**
+   * Google Play only. Optional boolean indicates personalized pricing on products available for purchase in the EU.
+   * For compliance with EU regulations. User will see "This price has been customized for you" in the purchase dialog when true.
+   * See https://developer.android.com/google/play/billing/integrate#personalized-price for more info.
+   */
+  googleIsPersonalizedPrice?: boolean | null;
+  /**
+   * iOS only, requires iOS 18.0 or greater with StoreKit 2. Win-back offer to apply to this purchase.
+   * Retrieve this using getEligibleWinBackOffersForPackage.
+   */
+  winBackOffer?: PurchasesWinBackOffer | null;
```

Comment:
I wondered about whether to add this... I see in RN, we added separate functions to make this mandatory in those cases... Is there a reason for that or would it be ok to make it an optional parameter like in here? @fire-at-will (This will also inform Flutter, where I didn't move it to `PurchaseParams` [initially](https://github.com/RevenueCat/purchases-flutter/commits/main/)... but I'm thinking we should..